include/ast_identifier.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef AST_IDENTIFIER
+#define AST_IDENTIFIER
+
+#include "ast_node.hpp"
+
+class Identifier : public Node {
+private:
+  std::string* identifier;
```

Comment:
Just chose to pass the pointer to `.cpp` and handle deallocation there instead of inside `.y` to make it less scary (stuff in `.y` already confused students) while also showing an example of a destructor use.